server/ngwmn/string_utils.py

```diff
@@ -0,0 +1,52 @@
+"""
+utility functions for the manipulation of strings
+
+"""
+
+
+def generate_subtitle(features):
+    """
+    logic to create subtitle -- which could also be called the 'monitoring location description'
+    :param features: python dictionary with features of the selected monitoring location
+    :return subtitle: the final subtitle string ready for display
+    """
+    site_type = features.get('SITE_TYPE')
+    aquifer_description = features.get('NAT_AQFR_DESC')
+    county = features.get('COUNTY_NM')
+    state = features.get('STATE_NM')
+
+    subtitle = ''
+
+    if site_type or aquifer_description:
```

Comment:
Codacy doesn't like all these nested if's. I think some of them could easily be removed by using the python ternary operator. You might also investigate using the string .format function rather than using string concatentation. You might want to think about this by forming pieces of the sentence and then stitch them together. Another place where you might make it more dry is in forming the county, state part of the string which is the same regardless of whether there is a site type or aquifer description.

Reply:
Great points! Sorry, about missing the Codacy check, I swear that all the lights were green on the checks before I added you to the review. Anyway, I reworked the 'ifs' in about a thousand different ways. Not really sure if I improved the the readability of the code as much as I would have liked, but I did manage to reduce the nesting.